Cargo.toml

```diff
@@ -11,22 +11,22 @@ edition = "2021"
 rust-version = "1.67.0"
 
 [features]
-default = []
-cli = ["dep:clap"]
+default = ["std"]
+std = ["data-encoding/std"]
+cli = ["std", "dep:clap"]
 
 [[bin]]
 name = "stellar-strkey"
 path = "src/bin/stellar-strkey/main.rs"
 required-features = ["cli"]
 doctest = false
 
-[build_dependencies]
+[build-dependencies]
 crate-git-revision = "0.0.6"
 
 [dev-dependencies]
 proptest = "1.0.0"
 
 [dependencies]
-data-encoding = "2.6.0"
-thiserror = "1.0.36"
+data-encoding = { version = "2.6.0", default-features = false, features = ["alloc"] }
```

Comment:
This might be a hard question to answer without some experimentation, but do we need alloc? All of the inputs and outputs of the strkey library are constrained in size. We _should_ I think be able to do all the base32 enc/dec without allocating, since the sizes should all be relatively small and predictable. Wdyt?